pyvista/plotting/charts.py

```diff
@@ -4163,7 +4168,8 @@ def add_chart(self, *charts):
     def remove_chart(self, chart_or_index):
         """Remove a chart from the collection."""
         chart = self._charts[chart_or_index] if isinstance(chart_or_index, int) else chart_or_index
-        assert chart in self._charts
+        if not (chart in self._charts):
+            raise ValueError('chart_index not present.')
```

Comment:
Not sure about this one, because I suspect this can be a fancy chart object which would be less informative in the error message. I'm also not sure how an out-of-bounds chart index behaves here. Going to leave that one up to you, we can leave the original phrasing (with capitalisation perhaps). But the `not in` is preferable either way.
```suggestion
        if chart not in self._charts:
            raise ValueError(f'Chart {chart_or_index} not present in collection.')
```

Reply:
The previous line will raise an IndexError when the (integer) index is out of bounds, so in the ValueError here `chart_or_index` will always be a chart object. But I guess it should be fine to print it out like that, I don't know of any other way to present it more cleanly in that case?

Reply:
Thanks @dcbr :) So then we can change the error to say `{chart}` to be explicit. _Or_ just stick to the original approach of not specifying the chart. After all it will be clear from what the user called the function with. That might be cleaner then.

As I said I'm good either way :)

Reply:
For the sake of simplicity and closing out this PR, let's go with this:
```suggestion
        if chart not in self._charts:
            raise ValueError('chart_index not present in charts collection.')
```